### PR TITLE
Create usb-platform-upgrade.sh

### PR DESCRIPTION
**Pull request by marcheschi**
Adapt smartos platform upgrade to work with Danube Cloud
Add the ability to download tgz without checking md5
It uses the /tmp folder to store files 
Paolo

### CODE REVIEW

**Review by YanChii — CHANGES_REQUESTED** [review]

File: bin/eslib/usb-platform-upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,303 @@
+#!/bin/bash
+
+# https://github.com/calmh/smartos-platform-upgrade
+# Copyright (c) 2012-2016 Jakob Borg & Contributors
+# Distributed under the MIT License
+#
+# Modified to work on Danube cloud by Paolo Marcheschi
+#
+
+mkdir /zones/tmp
```
[the remark this answers is not in the document]
Comment (marcheschi) [reply bin/eslib/usb-platform-upgrade.sh:10]: ok

File: bin/eslib/usb-platform-upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+	exit -1
+fi
+
+if [[ ! -n $platform_url ]]; then
+    host=https://us-east.manta.joyent.com
+    latest_path="${host}$(_curl "$host/Joyent_Dev/public/SmartOS/latest")"
+    version="$(expr "$latest_path" : '.*\([0-9]\{8\}T[0-9]\{6\}Z\).*')"
+    latest_spec_path="$(_curl "$host/Joyent_Dev/public/SmartOS/$version")"
+    header="$(expr "$latest_spec_path" : '.*platform-release-\([0-9]\{8\}\)-.*')"
+    platform_url="$latest_path/platform-release-$header-$version.tgz"
```
[the remark this answers is not in the document]
Comment (marcheschi) [reply bin/eslib/usb-platform-upgrade.sh]: For the moment I removed the joyent url, but to use DC url we have to define a static latest path, in order to be sure to download the latest platform for upgrade

File: bin/eslib/usb-platform-upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                echo -n ", MBR label"
+                usb="/dev/dsk/${removables[0]}p1"
+        fi
+else
+        usb="$1"
+        echo -n " using $usb"
+fi
+echo "usb path= " $usb
+umount "$usb" 2>/dev/null
+mkdir usb
```
[the remark this answers is not in the document]
Comment (marcheschi) [reply bin/eslib/usb-platform-upgrade.sh:226]: to create the mount point

**Comment by marcheschi** [comment]
I tried the latest committed script on DC it works:
[root@m2c (pacs) /zones/PAOLO]# ./platform-upgrade -u https://download.danube.cloud/esdc/factory/platform/platform-20201105T132431Z.tgz -f
Brand= esdc
Actual kernel=  20201105T132431Z
New Version=  20201105T132431Z
Already on latest version (20201105T132431Z).
Downloading platform-20201105T132431Z.tgz... OK
md5sum= ||
Extracting latest platform... OK
Marking release version... OK
Checking current boot device... detected c1t0d0, GPT labelusb path=  /dev/dsk/c1t0d0s2
, mounted, OK
Updating platform on boot device... OK
Remounting boot device... OK
Verifying kernel checksum on boot device... OK
Verifying boot_archive checksum on boot device... OK
Activating new platform on /dev/dsk/c1t0d0s2... please wait...Finished: OK

Boot device upgraded. To do:

 1) Sanity check the contents of /tmp/tmp.e6aM9j/usb
 2) umount /dev/dsk/c1t0d0s2
 3) reboot
[root@m2c (pacs) /zones/PAOLO]#

File: bin/eslib/usb-platform-upgrade.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (YanChii) [reply bin/eslib/usb-platform-upgrade.sh:226]: Ah, I've missed that there's no `$` before `usb` :)

**Review by YanChii — APPROVED** [review]

**Pull request merged**